packages/cmd/src/CommandProcessor.ts

```diff
@@ -721,6 +733,122 @@ export class CommandProcessor {
 
     }
 
+    /**
+     * Print parameters in place for --dry-run in effect
+     * @private
+     * @param {IHandlerParameters} commandParameters
+     * @returns
+     * @memberof CommandProcessor
+     */
+    private showInputsOnly(response: IHandlerResponseApi, commandParameters: IHandlerParameters) {
+
+        /**
+         * Determine if we should display secure values.  If the ENV variable is set to true,
+         * then we will display the secure values.  If the ENV variable is set to false,
+         * then we will display the non-secure values.
+         *                                                             - GitHub Copilot AI
+         */
+        let showSecure = false;
+
+        // ZOWE_SHOW_SECURE_ARGS
+        const SECRETS_ENV = `${ImperativeConfig.instance.envVariablePrefix}${CommandProcessor.ENV_SHOW_SECURE_SUFFIX}`;
+        const env = (process.env[SECRETS_ENV] || "false").toUpperCase();
+
+        if (env === "TRUE" || env === "1") {
+            showSecure = true;
+        }
+
+        // if config exists and a layer exists
+        let useConfig = false;
+        this.mConfig?.layers.forEach((layer) => {
+            if (layer.exists) {
+                useConfig = true;
+            }
+        });
+
+        /**
+         * Determine if Zowe V2 Config is in effect.  If it is, then we will construct
+         * a Set of secure fields from its API.  If it is not, then we will construct
+         * a Set of secure fields from the `ConnectionPropsForSessCfg` defaults.
+         */
+        const secureInputs: Set<string> =
+            useConfig ?
+                new Set(this.mConfig.api.secure.secureFields().map(v => v.split('.').slice(-1)[0])) :
+                new Set([...LoggerUtils.CENSORED_OPTIONS, ...LoggerUtils.SECURE_PROMPT_OPTIONS]);
```

Comment:
This one is tricky.  I spent some time with this to get it to work, and ideally, I'd like this censor logic to be in one place.  

`LoggerUtils.isSpecialValue` has a copied list of fields from [here](https://github.com/zowe/imperative/blob/master/packages/rest/src/session/ConnectionPropsForSessCfg.ts#L294).  This copied list is what we fall back to for v1 so we "support"  `--show-inputs-only` for both versions.  

I'm not clear if `LoggerUtils.isSpecialValue` is a generic API.  It adds a `.` to the special property name:
```typescript
        for (const v of specialValues) {
            if (prop.endsWith(`.${v}`)) return true;
        }
```
I'm also unclear where but with debug printing messages, I get `protocol` in the output when I print `specialValues`:
```
[
  'user',
  'password',
  'tokenValue',
  'keyPassphrase',
  'token-value',
  'base-path',
  'protocol',
  'token'
]
```

Unless there is something obvious I'm missing, I'd like to leave reconciling the secure arrays as a future tech debt item to resolve apart from this enhancement. 

Side note, idk if I'm reading this correctly, but is this logic correct?:
```typescript
            if (sec && typeof sec !== "object" && !LoggerUtils.isSpecialValue(prop))
                newData = newData.replace(new RegExp(sec, "gi"), LoggerUtils.CENSOR_RESPONSE);
```

The last condition reads like: "if not is special value then replace data"?  It seems like it would be, "if is special value then censor".

Reply:
> It adds a . to the special property name

that's v2 only, for v1 I guess we could replace the . with a _ 🙂
> I'd like to leave reconciling the secure arrays as a future tech debt item to resolve apart from this enhancement

That's fine by me (though we may never get to it) 😛
> The last condition reads like: "if not is special value then replace data"? It seems like it would be, "if is special value then censor".

Definitely feels like a bug unless the isSpecialValue give backward results 😛

Reply:
Thanks for reviewing @zFernand0 !